smartsim/settings/dragonRunSettings.py

```diff
@@ -0,0 +1,208 @@
+# modification, are permitted provided that the following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+# FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from __future__ import annotations
+
+import datetime
+import typing as t
+
+from ..log import get_logger
+from .base import RunSettings
+
+logger = get_logger(__name__)
+
+
+class DragonRunSettings(RunSettings):
+    def __init__(
+        self,
+        exe: str,
+        exe_args: t.Optional[t.Union[str, t.List[str]]] = None,
+        run_args: t.Optional[t.Dict[str, t.Union[int, str, float, None]]] = None,
+        env_vars: t.Optional[t.Dict[str, t.Optional[str]]] = None,
+        alloc: t.Optional[str] = None,
+        **kwargs: t.Any,
+    ) -> None:
+        """Initialize run parameters for a Dragon process
+
+        ``DragonRunSettings`` should only be used on systems where Dragon
+        is available.
+
+        If an allocation is specified, the instance receiving these run
+        parameters will launch on that allocation.
+
+        :param exe: executable to run
+        :type exe: str
+        :param exe_args: executable arguments, defaults to None
+        :type exe_args: list[str] | str, optional
+        :param run_args: srun arguments without dashes, defaults to None
+        :type run_args: dict[str, t.Union[int, str, float, None]], optional
+        :param env_vars: environment variables for job, defaults to None
+        :type env_vars: dict[str, str], optional
+        :param alloc: allocation ID if running on existing alloc, defaults to None
+        :type alloc: str, optional
+        """
+        super().__init__(
+            exe,
+            exe_args,
+            run_command="",
+            run_args=run_args,
+            env_vars=env_vars,
+            **kwargs,
+        )
+        self.alloc = alloc
+        self.mpmd: t.List[RunSettings] = []
+
+    reserved_run_args = {"chdir", "D"}
```

Comment:
Should we make this immutable or turn it into a property returning a `Mapping[str, str]`

Reply:
That is a good catch for other `RunSettings`-derived classes. Here we can just remove it.